mp/src/game/client/neo/ui/neo_hud_ghost_uplink_state.cpp

```diff
@@ -0,0 +1,101 @@
+#include "cbase.h"
+#include "neo_hud_ghost_uplink_state.h"
+
+#include "iclientmode.h"
+#include <vgui/ISurface.h>
+
+#include "c_neo_player.h"
+#include "weapon_ghost.h"
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+using vgui::surface;
+
+DECLARE_NAMED_HUDELEMENT(CNEOHud_GhostUplinkState, neo_ghost_uplink_state);
+
+NEO_HUD_ELEMENT_DECLARE_FREQ_CVAR(GhostUplinkState, 0.1)
+
+CNEOHud_GhostUplinkState::CNEOHud_GhostUplinkState(const char *pElementName, vgui::Panel *parent)
+	: CHudElement(pElementName), Panel(parent, pElementName)
+{
+	SetAutoDelete(true);
+
+	SetScheme("ClientScheme.res");
+
+	if (parent) {
+		SetParent(parent);
+	}
+	else {
+		SetParent(g_pClientMode->GetViewport());
+	}
+
+	for (int i = 0; i < numUplinkTextures; i++) {
+		m_pUplinkTextures[i] = surface()->CreateNewTextureID();
+		Assert(m_pUplinkTextures[i] > 0);
+		char textureFilePath[24];
+		V_sprintf_safe(textureFilePath, "%s%i", "vgui/hud/ctg/g_gscan_0", i);
+		surface()->DrawSetTextureFile(m_pUplinkTextures[i], textureFilePath, 1, false);
+	}
+
+	surface()->DrawGetTextureSize(m_pUplinkTextures[0], m_uplinkTexWidth, m_uplinkTexHeight);
+
+	SetVisible(true);
+}
+
+void CNEOHud_GhostUplinkState::ApplySchemeSettings(vgui::IScheme* pScheme)
+{
+	vgui::IScheme* scheme = vgui::scheme()->GetIScheme(vgui::scheme()->GetDefaultScheme());
+	Assert(scheme);
+
+	int wide, tall;
+	surface()->GetScreenSize(wide, tall);
+	int centerX = wide / 2;
+	int centerY = tall / 2;
+	textureXPos = centerX - (m_uplinkTexWidth / 2);
+	constexpr int compassHeightPlusMargins = 30;
+	textureYPos = tall - m_uplinkTexHeight - compassHeightPlusMargins;
+	SetBounds(textureXPos, textureYPos, m_uplinkTexWidth, m_uplinkTexHeight);
+
+	BaseClass::ApplySchemeSettings(pScheme);
+}
+
+void CNEOHud_GhostUplinkState::UpdateStateForNeoHudElementDraw()
+{
+}
```

Comment:
Texture change could be utilised in this function instead.

Reply:
Then these need to be member variables
```
static constexpr int numTextureStartTimes = 11;
int textureOrder[numTextureStartTimes] = {0, 1, 2, 3, 2, 0, 1, 2, 3, 0, 1};
float textureStartTimes[numTextureStartTimes] = {0, 0.2, 0.5, 0.6, 1.0, 1.1, 1.4, 1.8, 2.5, 2.7, 3.0};
```

Reply:
Still not seeing how, instead of going to `DrawNeoHudElement` it can go split into two of those functions?

Reply:
I guess I should wrap them together since they're related

Reply:
But I don't know how so I'l just leave them like this

Reply:
IMO how I'll split it:
* `textureStartTimes` -> local variable in `UpdateStateForNeoHudElementDraw`
* `textureOrder` -> local variable in `DrawNeoHudElement`
* `NUM_TEXTURE_START_TIMES` -> static/anonymous-namespaced var within `neo_hud_ghost_uplink_state.cpp`